### PR TITLE
refactor(app): update app settings advanced tab for OT-3 desktop

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

OT-3 desktop updates for the app settings page, including changing the default "Share Robot and App Analytics with Opentrons" to true. Creates a config value for toggling on agnostic OT-2/OT-3 protocol analysis, and a general "Enable Extended Hardware" feature flag to hide the agnostic OT-2/OT-3 protocol analysis section and any future OT-3 desktop updates that would otherwise be visible to current OT-2 users.

closes RCORE-233, closes RCORE-237

# Changelog

 - Updates app settings advanced tab for OT-3 desktop
 - Changes analytics default to opt-in

# Review requests

confirm match to designs, config/feature flag approach

# Risk assessment

low

### CODE REVIEW

File: app/src/atoms/buttons/ToggleButton.tsx [hunk 1]
```diff
@@ -35,12 +37,30 @@ const TOGGLE_ENABLED_STYLES = css`
...
+      aria-label={label}
+      aria-checked={toggledOn}
       size={SIZE_2}
       css={props.toggledOn ? TOGGLE_ENABLED_STYLES : TOGGLE_DISABLED_STYLES}
-      {...props}
-    />
+      {...buttonProps}
+    >
+      {/* TODO(bh, 2022-10-05): implement small and large sizes from design system */}
+      <Icon name={iconName} height="1rem" />
```
Comment (koji) [inline app/src/atoms/buttons/ToggleButton.tsx]: maybe we can use `SIZE_1` instead of 1rem

**Review by koji — APPROVED** [review]
added feature flag works as expected and the changes look good to me.
One thing that I want to know about this change. 
The design in Figma doesn't consider the `Confirm Heater-Shaker Module Attachment` section.

I'm curious about the order that the design team expected.

**Comment by brenthagen** [comment]
> added feature flag works as expected and the changes look good to me. One thing that I want to know about this change. The design in Figma doesn't consider the `Confirm Heater-Shaker Module Attachment` section.
> 
> I'm curious about the order that the design team expected.

that section was added in 6.1: https://www.figma.com/file/695ogls23YsQYImRcGYDE1/6.1-Heater%2FShaker?node-id=5690%3A93333

looks like it should be between `USB-to-Ethernet Adapter Information` and `Show Link to Get Labware Offset Data`

**Comment by koji** [comment]
> > added feature flag works as expected and the changes look good to me. One thing that I want to know about this change. The design in Figma doesn't consider the `Confirm Heater-Shaker Module Attachment` section.
> > I'm curious about the order that the design team expected.
> 
> that section was added in 6.1: https://www.figma.com/file/695ogls23YsQYImRcGYDE1/6.1-Heater%2FShaker?node-id=5690%3A93333
> 
> looks like it should be between `USB-to-Ethernet Adapter Information` and `Show Link to Get Labware Offset Data`

Thank you.
I just checked v6.1.0 app. Somehow the order issue passed Design Q&A.

Screen Shot 2022-10-12 at 3 13 30 PM


Could you include this order issue fix in this PR?

**Pull request merged**